events/bingo/bingoBossApp/ui/js/bingoBossApp_ui.js

```diff
@@ -0,0 +1,67 @@
+//
+//  bingoBossApp_ui.js
+//
+//  Created by Zach Fox on 2019-02-15
+//  Copyright 2019 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+
+/* globals $ document EventBridge setTimeout */
+
+// Emit an event specific to the `bingoBossApp` over the EventBridge.
+function emitBingoBossEvent(type) {
+    var event = {
+        app: 'bingoBossApp',
+        type: type
+    };
+    EventBridge.emitWebEvent(JSON.stringify(event));
+}
+
+
+// Handle incoming events over the EventBridge.
+// Possible events include updating the "status text" area of the Boss app.
+function onScriptEventReceived(scriptEvent) {
+    try {
+        scriptEvent = JSON.parse(scriptEvent);
+    } catch (error) {
+        console.log("ERROR parsing scriptEvent: " + error);
+        return;
+    }
+
+    if (scriptEvent.app !== "bingoBossApp") {
+        return;
+    }
+
+    
+    switch (scriptEvent.method) {
+        case "updateStatus":
+            document.getElementById("statusText").innerHTML = scriptEvent.statusText;
+            break;
+
+        default:
+            console.log("Unknown message from bingoBossApp_app.js: " + JSON.stringify(scriptEvent));
+            break;
+    }
+}
+
+
+// This delay is necessary to allow for the JS EventBridge to become active.
+// The delay is still necessary for HTML apps in RC78+.
+var EVENTBRIDGE_SETUP_DELAY = 500;
+function onLoad() {
+    setTimeout(function() {
+        EventBridge.scriptEventReceived.connect(onScriptEventReceived);
+        emitBingoBossEvent("eventBridgeReady");
+    }, EVENTBRIDGE_SETUP_DELAY);
+

```

Comment:
Maybe just use regular is instead of jquery and remove the library, unless it makes your life easier for this event handler.  I think it was the only jquery use.

Reply:
great call

Reply:
actually, now that i think about this, i don't want to introduce such a functional change this late in the game. this is a great idea for the future. the reason this still uses jQuery is because that's what it used in the beginning and i didn't have time to remove it.

thank you for this comment though, this is a great point